modules/altinnsalary/view/salarylist.php

```diff
@@ -310,5 +316,78 @@
   }
 ?>
 
+<script type="text/javascript">
+  var salary_virksomhet_array = <?= json_encode($salary_virksomhet_array); ?>;
+  var work_relation_virksomhet_array = <?= json_encode($workrelation_virksomhet_array); ?>;
+  var salary_selected = [];
+  var work_relations_selected = [];
+  var virksomhet_names = <?= json_encode($virksomhet_names); ?>;
+  var virksomhet_otp_amounts = [];
+  for(var virksomhet_id in virksomhet_names) {
+    virksomhet_otp_amounts[virksomhet_id] = 0;
+  }
+
+  function refreshSelectedWorkRelations() {
```

Comment:
Instead of looping though all, pass an element and affect the object that keeps track of what is selected and call the function to render current subunits' OTP, same for salaries.